### DISCUSSION TITLE
How to mount folder from one GCS bucket to another as virtual folder.

### DISCUSSION THREAD

**Discussion by rashmimaharana** [drakkan/sftpgo#508] [General]
I would like mount a folder as virtual from one GCS bucket to another. Currently only local filesystem is working. When i tried to mount one from GCS, it gives below error. Any possible workaround.

Error :
\n\tgithub.com/go-chi/chi/v5@v5.0.3/mux.go:88 +0x310\nnet/http.serverHandler.ServeHTTP(0xc00028a0e0, 0x1b2b758, 0xc0000241c0, 0xc0005b3200)\n\tnet/http/server.go:2887 +0xa3\nnet/http.(*conn).serve(0xc000918780, 0x1b32d48, 0xc000823dc0)\n\tnet/http/server.go:1952 +0x8cd\ncreated by net/http.(*Server).Serve\n\tnet/http/server.go:3013 +0x39b\n","panic":"runtime error: invalid memory address or nil pointer dereference"}

**Comment by drakkan** [comment]
Hi,

this looks like a bug, can you please share the exact steps to reproduce it and the SFTPGo version you are using? Thanks

**Reply by rashmimaharana** [reply]
Setup :
1> Run cloudSQL postgres in GCP
2> Run SFTPGo : sftpgo_v2.1.0_linux_x86_64.tar.xz in GCP VM. Provided sufficient roles to Service Account attached to VM to manage GCS bucket creation.
 
Steps to produce:
1> Create one GCS bucket in GCP , example : "test-project-gcp-bucket-us-cental-1"
2> Create 2 directory/folder in that bucket named "transfer" and "processed"
3> Create one user account with below details :
                > user name : USER_1
                > password : password
                > permissions : * 
                > chose storage as GCS 
                > check "Automatic Credential" ON
                > give the bucket name : "test-project-gcp-bucket-us-cental-1"
                > key prefix : transfer/

Now upload some files to user account, those files will go to GCS bucket : "gs://test-project-gcp-bucket-us-cental-1/transfer" . It will land in directory "transfer/" .
**Now i would like to mount that directory as virtual folder in different user account which will have their own GCS bucket and the above shared virtual folder. To do that I proceeded like below.**

4> Now open SFTPGo UI, click on folder section and create a virtual folder from same GCS bucket : gs://test-project-gcp-bucket-us-cental-1
                > name : virtual-folder-1
                > storage : Google Cloud Storage
                > Bucket : "test-project-gcp-bucket-us-cental-1"
                > Check Automatic Credentials ON
                > key prefix : transfer/
This step is giving me the error. Its not allowing me to create a virtual folder object using SFTPGo UI even though GCS bucket and directory created before hand  (Ex : gs://test-project-gcp-bucket-us-cental-1/transfer).
I would like to share the directory "gs://test-project-gcp-bucket-us-cental-1/transfer" in different user account, so that other account will have their own bucket say "gs://test-project-gcp-bucket-us-cental-2/transfer" and also a shared directory from above.

**Reply by rashmimaharana** [reply]
<img width="935" alt="screen" src="https://user-images.githubusercontent.com/10846216/130031978-1db4aa73-48e3-4990-ae22-6c36996554ff.PNG">

**Comment by drakkan** [comment]
Hi,

this bug was fixed [some days ago](https://github.com/drakkan/sftpgo/issues/510). I would appreciate a feedback to confirm that the issue is fixed for you too (I tested locally and not using a GCP VM, this should make no difference), thank you

**Reply by rashmimaharana** [reply]
Thanks @drakkan 
Its working fine now. Thanks for the help :)
